pkg/server/plugin/keymanager/gcpkms/client_fake.go

```diff
@@ -709,6 +738,7 @@ func (k *fakeKMSClient) createFakeCryptoKeyVersion(cryptoKey *kmspb.CryptoKey, v
 		},
 		CryptoKeyVersion: &kmspb.CryptoKeyVersion{
 			Name:      path.Join(cryptoKey.Name, "cryptoKeyVersions", version),
+			State:     k.initialCryptoKeyVersionState,
```

Comment:
what is this `initialCryptoKeyVersionState` change accomplishing?

Reply:
So, for testing the scenario in which we keep retrying the retrieval of the public key when the crypto key state is still in `CryptoKeyVersion_PENDING_GENERATION` state, we need to be able to mock this initial state, and since we weren't doing it, every crypto key created by our test was starting with the default value `CryptoKeyVersion_CRYPTO_KEY_VERSION_STATE_UNSPECIFIED`, making it impossible to reach the desired scenario.

Reply:
The retry that I mentioned can be seem here: pkg/server/plugin/keymanager/gcpkms/gcpkms.go:1056